baybe/recommenders/pure/bayesian/sequential_greedy.py

```diff
@@ -1,275 +1,20 @@
-        self, _: Any, value: float
-    ) -> None:
-        """Validate that the given value is in fact a percentage.
-
-        Raises:
-            ValueError: If ``value`` is not between 0 and 1.
-        """
-        if not 0 <= value <= 1:
-            raise ValueError(
-                f"Hybrid sampling percentage needs to be between 0 and 1 but is {value}"
-            )
-
-    def _recommend_discrete(
-        self,
-        subspace_discrete: SubspaceDiscrete,
-        candidates_comp: pd.DataFrame,
-        batch_size: int,
-    ) -> pd.Index:
-        """Generate recommendations from a discrete search space.
-
-        Args:
-            subspace_discrete: The discrete subspace from which to generate
-                recommendations.
-            candidates_comp: The computational representation of all discrete candidate
-                points to be considered.
-            batch_size: The size of the recommendation batch.
-
-        Raises:
-            NoMCAcquisitionFunctionError: If a non-Monte Carlo acquisition function is
-                used with a batch size > 1.
-
-        Returns:
-            The dataframe indices of the recommended points in the provided
-            computational representation.
-        """
-        # For batch size > 1, this optimizer needs a MC acquisition function
-        if batch_size > 1 and not self.acquisition_function.is_mc:
-            raise NoMCAcquisitionFunctionError(
-                f"The '{self.__class__.__name__}' only works with Monte Carlo "
-                f"acquisition functions for batch sizes > 1."
-            )
-
-        from botorch.optim import optimize_acqf_discrete
-
-        # determine the next set of points to be tested
-        candidates_tensor = to_tensor(candidates_comp)
-        points, _ = optimize_acqf_discrete(
-            self._botorch_acqf, batch_size, candidates_tensor
-        )
-
-        # retrieve the index of the points from the input dataframe
-        # IMPROVE: The merging procedure is conceptually similar to what
-        #   `SearchSpace._match_measurement_with_searchspace_indices` does, though using
-        #   a simpler matching logic. When refactoring the SearchSpace class to
-        #   handle continuous parameters, a corresponding utility could be extracted.
-        idxs = pd.Index(
-            pd.merge(
-                candidates_comp.reset_index(),
-                pd.DataFrame(points, columns=candidates_comp.columns),
-                on=list(candidates_comp),
-            )["index"]
+class SequentialGreedyRecommender(BotorchRecommender):
+    """A :class:`baybe.recommenders.pure.bayesian.botorch.BotorchRecommender` alias for backward compatibility."""  # noqa: D401, E501
+
+    def __attrs_pre_init__(self):
+        warnings.warn(
+            f"The class `SequentialGreedyRecommender` has been deprecated and will be "
+            f"removed in a future version. Please use `{BotorchRecommender.__name__}` "
```

Comment:
technically this should mention that the sequential flag must be True for a 1to1 replacement

Reply:
Technically: no, in fact :D Because the previous behavior WAS the one with `sequential=True` (i.e. bug). That is, you'd really get the old behavior when doing just nothing and using the `sequential=False` default. Makes sense?

Reply:
to follow up: imo this is both
1) a refactoring, renaming the recommender name
2) a bug fix

Imo, we do not need to mix 1 and 2, the error should not carry over into the deprecation.

In fact the README should have an entry in Fixed that says seq greedy recommender sequential flag not being activated

Reply:
I know what you mean, and if you really prefer that, I will not block it (have pushed a commit that you can inspect and see if that's what you want). But the downsides I see about this is that the consequences could be difficult to disentangle for the user. With my previous approach, they wouldn't need to bother: upgrading to `BotorchRecommender` would really give them the same results as before, and everything becomes automatically bug-free. With your approach, on the other hand:
* They would see that they need to upgrade, but the deprecation message would explicitly tell them to active `sequential=True`
* This would NOT be a 1:1 replacement to what they had before, giving different results.
* Instead, they would also need to know (magically, unless they look into the changelog, by chance) that `sequential=True` is a bugfix and not actually what they need to do for backward-compatibility.

In fact, when I read the new deprecation warning myself, I was also a bit confused first 😄 

So, in a sense, you are asking them to break their own backward-compatibility, and the changelog mentions a bug that is no longer relevant to the users, since they would anyway upgrade to `BotorchRecommender` as the next step. With the previous approach, we'd simply skip that "inconsistent stage" altogether, keeping the old behavior but now with a class label that no longer makes it a bug.

Reply:
the downside of your proposed way
- providing the user a `SequentialGreedyRecommender` that is not sequential

seems to outweigh all your points and I dont see why providing a deprecation 1to1 replacement for something should take precedence over providing `what the user has requested when they wrote the word Sequential`. Given that we have no evidence of any more computational expense or so, what do you mean with `user has to bother`?

I see that calling it a `1to1` replacement doesnt make sense though, just use `please use BotorchRecommender(sequential=True)` or similar

Reply:
As I said, I'm fine with your proposal because it's a deprecation anyway, so I "accept" :D But I in fact noticed a much more important issue – and I think that is the reason why we called it `sequential` back at times when we only supported discrete spaces: the `sequential` flag is relevant only for conti spaces. For discrete/mixed spaces, it automatically performs sequential optimization, whereas for conti spaces, sequential is `False` by default.

Given the above, I think it's confusing to just expose `sequential=False` as default, as it would be quite misleading for non-conti spaces. Would you be fine if I instead set it to `sequential: bool | None = None` and also raise the corresponding warnings at runtime (e.g. warning the user that his `sequential` flag being set to `False` will be ignored once he uses the recommender in a non-conti setting)? Will prepare the change and you can then simply review it

Reply:
Updated logic is now there. Please let me know what you think / resolve

Reply:
if `sequential` only applies to discrete spaces, why dont we make that clear in the flag name and make the logic even simpler (ie not so many warnings or explanations needed)?

Reply:
Also OK for me. Do you have a suggestion?

Reply:
`sequential_discrete`

Reply:
sorry I prob misread, if its only relevant in conti spaces it should be `sequential_continuous`, right?

Reply:
Yeah, noticed it, too :D Now I've fixed it